examples/list.rs

```diff
@@ -1,24 +1,70 @@
-use std::{
-    error::Error,
-    io,
-    time::{Duration, Instant},
-};
+use std::{error::Error, io};
 
 use crossterm::{
     event::{self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, KeyEventKind},
     execute,
     terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
 };
-use ratatui::{prelude::*, widgets::*};
+use ratatui::{prelude::*, style::palette::tailwind, widgets::*};
+
+const TODO_HEADER_BG: Color = tailwind::BLUE.c950;
+const NORMAL_ROW_COLOR: Color = tailwind::SLATE.c950;
+const ALT_ROW_COLOR: Color = tailwind::SLATE.c900;
+const SELECTED_STYLE_FG: Color = tailwind::BLUE.c300;
+const TEXT_COLOR: Color = tailwind::SLATE.c200;
+const COMPLETED_TEXT_COLOR: Color = tailwind::GREEN.c600;
+
+enum Status {
+    Todo,
+    Completed,
+}
+
+struct TodoItem {
+    todo: String,
+    info: String,
+    status: Status,
+}
+
+impl TodoItem {
+    fn styled_list_item(&self, index: usize) -> ListItem {
+        let bg_color = match index % 2 {
+            0 => NORMAL_ROW_COLOR,
+            _ => ALT_ROW_COLOR,
+        };
+        let (style, todo_str) = match self.status {
+            Status::Completed => (
+                Style::default().fg(COMPLETED_TEXT_COLOR).bg(bg_color),
+                " ✓ ".to_string() + &self.todo,
+            ),
+            Status::Todo => (
+                Style::default().fg(TEXT_COLOR).bg(bg_color),
+                " ☐ ".to_string() + &self.todo,
+            ),
+        };
+        ListItem::new(Line::from(todo_str)).style(style)
```

Comment:
Does something like this read a bit clearer?

```suggestion
        let line = match self.status {
            TodoState::Todo =>
                Line::styled(format!(" ☐ {}", self.todo, (TEXT_COLOR, bg_color))),
            TodoState::Done =>
                Line::styled(format!(" ✓ {}", self.todo), (COMPLETED_TEXT_COLOR, bg_color)),
        };
        ListItem::new(line)
```

Or does the style really need to be on the list item to make this look good?